docs/tutorials/schrödinger_equation.rst

```diff
@@ -0,0 +1,271 @@
+    rhs(jnp.array(0.1), jnp.array([0., 1.]))
+
+We eventually want to find the probability of existence of this qubit state, so we will prepare the following function.
+
+.. jupyter-execute::
+
+    def state_probabilities(state):
+        return unp.abs(state) ** 2
+
+3. Solve the right-hand side function using existing solvers
+------------------------------------------------------------
+
+Here we show how the rhs function can be passed to numerical ODE solvers in both SciPy and JAX as if the function had been natively written in either library.
+
+First, we solve the equation by using Numpy as the input and ``scipy.integrate.solve_ivp`` as a solver.
+We define the initial state, the time span for the simulation, and time point we want to simulate.
+
+.. jupyter-execute::
+
+    init_state = np.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = np.linspace(0,(N-1) * dt,N)
+
+We solve by using ``scipy.integrate.solve_ivp`` and plot the probabilities of each state.
+
+.. jupyter-execute::
+
+    sol = solve_ivp(rhs,t_span,init_state,method='RK45',t_eval=T)
+    probabilities = state_probabilities(sol.y)
+
+    plt.plot(sol.t, probabilities[0], label="0")
+    plt.plot(sol.t, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+Second, we solve the equation by using Jax.array as the input and ``jax.experimental.ode.odeint`` as a solver.
+
+.. jupyter-execute::
+
+    init_state = jnp.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = jnp.linspace(0,(N-1) * dt,N)
+
+    sol = odeint(lambda y,t: rhs(t,y), init_state, T)
+    probabilities = state_probabilities(sol.T)
+    plt.plot(T, probabilities[0], label="0")
+    plt.plot(T, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+
+4. Register custom solvers using :meth:`.Alias.register_function`
+-----------------------------------------------------------------
+
+In this section, we introduce how to register the function, taking the 4th order Runge-Kutta method
+to solve ordinary differential equations as an example.
+
+The 4th order Runge-Kutta fomula is here:
+
+.. math::
+
+    k_1 &= h \cdot f(t_n, y_n) \\
+    k_2 &= h \cdot f(t_n + \frac{h}{2}, y_n + \frac{k_1}{2}) \\
+    k_3 &= h \cdot f(t_n + \frac{h}{2}, y_n + \frac{k_2}{2}) \\
+    k_4 &= h \cdot f(t_n + h, y_n + k_3) \\
+    y_{n+1} &= y_n + \frac{1}{6}(k_1 + 2k_2 + 2k_3 + k_4)
+
+, where :math:`y_{n}`, :math:`t_{n}`, and :math:`h` are current solution, current time, and time step size, respectively.
+
+We define the Runge-Kutta method to be used later here:
+
+.. jupyter-execute::
+
+    def runge_kutta_step(n, state):
+        k1 = dt * rhs(n * dt, state)
+        k2 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k1)
+        k3 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k2)
+        k4 = dt * rhs(n * dt + dt, state + k3)
+        return (k1 + 2*k2 + 2*k3 + k4) / 6.
```

Comment:
I wonder if here the method should be defined to take in the state, current time, time step size, and rhs. E.g.

```
def runge_kutta_step(t, y, dt, rhs):
    k1 = dt * rhs(t, y)
    k2 = dt * rhs(t + 0.5 * dt, y + 0.5 * k1)
    ...
```